File: 2025/Company/Company.Web/Modules/ServerTypes/Company/BusinessEntityContactColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BusinessEntityContactRow } from "./BusinessEntityContactRow";

export interface BusinessEntityContactColumns {
    BusinessEntityId: Column<BusinessEntityContactRow>;
    PersonType: Column<BusinessEntityContactRow>;
    ContactTypeName: Column<BusinessEntityContactRow>;
}

export class BusinessEntityContactColumns extends ColumnsBase<BusinessEntityContactRow> {
    static readonly columnsKey = 'Company.BusinessEntityContact';
    static readonly Fields = fieldsProxy<BusinessEntityContactColumns>();
}

File: 2025/Company/Company.Web/Modules/Company/BusinessEntityContact/BusinessEntityContactColumns.cs
namespace Company.Company.Columns;

[ColumnsScript("Company.BusinessEntityContact")]
[BasedOnRow(typeof(BusinessEntityContactRow), CheckNames = true)]
public class BusinessEntityContactColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int BusinessEntityId { get; set; }
    public string PersonType { get; set; }
    public string ContactTypeName { get; set; }
}